persist-cli/src/main/java/io/ballerina/persist/utils/BalProjectUtils.java

```diff
@@ -691,4 +699,81 @@ public static void validatePullCommandOptions(String datastore, String host, Str
         }
     }
 
+    public static void validateDatastore(String datastore) throws BalException {
+        if (!SUPPORTED_DB_PROVIDERS.contains(datastore)) {
+            throw new BalException(String.format("the persist layer supports one of data stores: %s" +
+                    ". but found '%s' datasource.", Arrays.toString(SUPPORTED_DB_PROVIDERS.toArray()), datastore));
+        }
+    }
+
+    public static void validateTestDatastore(String datastore, String testDatastore) throws BalException {
+        if (testDatastore == null) {
+            return;
+        }
+
+        if (!SUPPORTED_TEST_DB_PROVIDERS.contains(testDatastore)) {
+            throw new BalException(String.format("the persist layer supports one of test data stores: %s. but found " +
+                    "'%s' datasource.", Arrays.toString(SUPPORTED_TEST_DB_PROVIDERS.toArray()), testDatastore));
+        }
+
+        if (testDatastore.equals(IN_MEMORY_TABLE) && !SUPPORTED_NOSQL_DB_PROVIDERS.contains(datastore)) {
+            throw new BalException(String.format("in-memory table is the only supported test data store for " +
+                    "nosql data stores. but found '%s' datasource.", datastore));
+        }
+
+        if (testDatastore.equals(H2_DB) && !SUPPORTED_SQL_DB_PROVIDERS.contains(datastore)) {
+            throw new BalException(String.format("H2 is the only supported test data store for " +
+                    "sql data stores. but found '%s' datasource.", datastore));
```

Comment:
Same as above,

`The H2 is supported as the test data store for SQL data stores. but found '%s' data source.`